pkg/mpc-examples/contracts/04-pause-unpause-controller/PauseUnpauseControllerFactory.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/IManagedPoolFactory.sol";
+
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IManagedPool.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Create2.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./PauseUnpauseController.sol";
+
+/**
+ * @title PauseUnpauseControllerFactory
+ * @notice A factory contract able to create controller contract for Balancer's managed pools.
+ */
+contract PauseUnpauseControllerFactory is Ownable {
+    mapping(address => bool) public isControllerFromFactory;
+
+    IManagedPoolFactory public immutable managedPoolFactory;
+    bool private _disabled;
+    IVault public immutable balancerVault;
+
+    uint256 private _nextControllerSalt;
+    address private _lastCreatedPool;
+
+    // This struct is a subset of IManagedPoolFactory.NewPoolParams which omits arguments
+    // that this factory will override and are therefore unnecessary to provide. It will
+    // ultimately be used to populate IManagedPoolFactory.NewPoolParams.
+    struct MinimalPoolParams {
+        string name;
+        string symbol;
+        IERC20[] tokens;
+        uint256[] normalizedWeights;
+        uint256 swapFeePercentage;
+        bool swapEnabledOnStart;
+        uint256 managementAumFeePercentage;
+        uint256 aumFeeId;
+    }
+
+    event ControllerCreated(address indexed controller, bytes32 poolId);
+    event Disabled();
+
+    constructor(IVault vault, address factory) {
+        managedPoolFactory = IManagedPoolFactory(factory);
+        balancerVault = vault;
+    }
+
+    /// === Getters === ///
+
+    function isDisabled() public view returns (bool) {
+        return _disabled;
+    }
```

Comment:
I know you're trying to group your getters, but it might be cleaner to group this with the rest of the enable/disable functions at the bottom. Also, the NullController now implements this differently, also validating that the pool factory itself is enabled:
```

    function isDisabled() external view returns (bool) {
        return _disabled || _isPoolFactoryDisabled();
    }

    /**
     * @dev Query whether the pool factory is disabled.
     */
    function _isPoolFactoryDisabled() internal view returns (bool) {
        return IManagedPoolFactory(managedPoolFactory).isDisabled();
    }
```